cmake/modules/BuildJaeger.cmake

```diff
@@ -45,10 +45,12 @@ function(build_jaeger)
   if(CMAKE_MAKE_PROGRAM MATCHES "make")
     # try to inherit command line arguments passed by parent "make" job
     set(make_cmd $(MAKE))
+  elseif(CMAKE_MAKE_PROGRAM MATCHES "ninja")
+    set(make_cmd ninja)
   else()
-    set(make_cmd ${CMAKE_COMMAND} --build <BINARY_DIR> --config $<CONFIG> --target Jaeger)
+    set(make_cmd ${CMAKE_MAKE_PROGRAM} --build <BINARY_DIR> --config $<CONFIG> --target Jaeger)
```

Comment:
i still don't understand why this change is necessary or what is the "nature of ninja build" different from that of cmake or make.

Reply:
@tchaikov we are setting make_cmd as $(MAKE) if we use cmake but when we are using ninja it expects : `ninja <target_name> ` I suppose ninja does not support `$(MAKE)` and use parallel cores by default.
```
$ ./do_cmake.sh -DWITH_JAEGER=ON
$ cd build/
$ ninja
ninja: error: build.ninja:1225: bad $-escape (literal $ must be written as $$)
  COMMAND = cd /workdir/ceph/build/external/thrift && $(MAKE)...
                                                      ^ near here
```
https://tracker.ceph.com/issues/51029

Reply:
how about using `cmake` directly then?